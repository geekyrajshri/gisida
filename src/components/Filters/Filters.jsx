import React from 'react';
import PropTypes from 'prop-types';
import Filter from '../Filter/Filter';

require('./Filters.scss');

const Filters = ({ filters, headers }) =>
  (<ul className="filters">
    {filters.map(filter =>
      (<Filter
        key={filter}
        filter={filter}
        headers={headers}
      />))
    }
  </ul>);

Filters.propTypes = {
  filters: PropTypes.arrayOf(PropTypes.any).isRequired,
  headers: PropTypes.arrayOf(PropTypes.any),
};

Filters.defaultProps = {
  headers: [],
};

export default Filters;
