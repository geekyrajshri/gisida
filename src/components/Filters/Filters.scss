ul.filters {
  padding-left: 0.8em;
}
