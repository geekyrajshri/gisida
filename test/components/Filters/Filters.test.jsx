import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '../../../src/components/Filters/Filters';

describe('Filters', () => {
  const onToggleView = jest.fn();
  const onFilterSelect = jest.fn();
  const filters = ['test_filters'];
  const view = 'test_view';
  const headers = ['test_headers'];
  const sector = 'test_sector';
  const checked = ['test_checked'];
  const currentView = 'test_currentView';
  it('Filters renders each filters', () => {
    const component = renderer.create(
      <Filters
        onToggleView={onToggleView}
        onFilterSelect={onFilterSelect}
        filters={filters}
        view={view}
        headers={headers}
        sector={sector}
        checked={checked}
        currentView={currentView}
        mapTargetId={'map-1'}
      />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
