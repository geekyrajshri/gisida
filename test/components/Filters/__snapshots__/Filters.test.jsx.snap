// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Filters renders each filters 1`] = `
<ul
  className="filters"
  style={
    Object {
      "display": "none",
    }
  }
>
  <li
    className="filter"
  >
    <label
      htmlFor="test_filters"
    >
      <input
        checked={false}
        id="test_sector"
        name="test_sector"
        onClick={[Function]}
        type="radio"
        value="test_filters"
      />
      test_filters
    </label>
  </li>
</ul>
`;
